pkg/plugin/processor/builtin/convertField.go

```diff
@@ -0,0 +1,156 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=convertField_paramgen.go convertFieldConfig
+
+package builtin
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+type convertField struct {
+	referenceResolver sdk.ReferenceResolver
+	config            convertFieldConfig
+
+	sdk.UnimplementedProcessor
+}
+
+func newConvertField() *convertField {
+	return &convertField{}
+}
+
+type convertFieldConfig struct {
+	// Field is the target field, as it would be addressed in a Go template (e.g. `.Payload.After.foo`).
+	// you can only convert fields that are under .Key and .Payload, and said fields should contain structured data.
+	Field string `json:"field" validate:"required,regex=^.(Payload|Key).*"`
+	// Type is the target field type after conversion, available options are: string, int, float, bool.
+	Type string `json:"type" validate:"required,inclusion=string|int|float|bool"`
+}
+
+func (p *convertField) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "field.convert",
+		Summary: "Convert the type of a field.",
+		Description: `Convert takes the field of one type and converts it into another type (e.g. string to integer). 
+The applicable types are string, int, float and bool. Converting can be done between any combination of types. Note that
+booleans will be converted to numeric values 1 (true) and 0 (false). Processor is only applicable to .Key, .Payload.Before
+and .Payload.After prefixes, and only applicable if said fields contain structured data.
+if the record contains raw JSON data, then use the processor "decode.json" to parse it into structured data first.`,
+		Version:    "v0.1.0",
+		Author:     "Meroxa, Inc.",
+		Parameters: convertFieldConfig{}.Parameters(),
+	}, nil
+}
+
+func (p *convertField) Configure(ctx context.Context, m map[string]string) error {
+	err := sdk.ParseConfig(ctx, m, &p.config, convertFieldConfig{}.Parameters())
+	if err != nil {
+		return cerrors.Errorf("failed to parse configurations: %w", err)
```

Comment:
Nitpick: I'd maybe rather call it "configuration" (singular) instead of "configurations", I see it as one configuration that contains multiple values / parameters.